### PR TITLE
Fix CVE-2016-4978 in 2.4.x branch

### PR DESCRIPTION
**Pull request by gaol**
The fix in `2.3.25.x` is missing in `2.4.x` branch, See https://github.com/hornetq/hornetq/pull/2125

Issue: https://issues.redhat.com/browse/JBEAP-22562

### CODE REVIEW

**Comment by gaol** [comment]
@gaohoward would you please review? thanks

File: hornetq-jms-server/src/main/java/org/hornetq/jms/server/config/impl/ConnectionFactoryConfigurationImpl.java [hunk 1]
```diff
@@ -785,7 +797,12 @@ public int getEncodeSize()
 
          BufferHelper.sizeOfNullableSimpleString(groupID) +
```
Comment (gaohoward) [inline hornetq-jms-server/src/main/java/org/hornetq/jms/server/config/impl/ConnectionFactoryConfigurationImpl.java:798]: what's the '+' at the end?

File: hornetq-jms-server/src/main/java/org/hornetq/jms/server/config/impl/ConnectionFactoryConfigurationImpl.java [hunk 2]
```diff
@@ -785,7 +797,12 @@ public int getEncodeSize()
...
 
-         DataConstants.SIZE_INT; // factoryType
+         DataConstants.SIZE_INT +
+         // factoryType
+
+         BufferHelper.sizeOfNullableSimpleString(deserializationBlackList) +
```
Comment (gaohoward) [inline hornetq-jms-server/src/main/java/org/hornetq/jms/server/config/impl/ConnectionFactoryConfigurationImpl.java:803]: same '+' at the end. just make sure it's not added accidentally in to the code

File: hornetq-jms-server/src/main/java/org/hornetq/jms/server/config/impl/ConnectionFactoryConfigurationImpl.java [hunk 3]
```diff
@@ -785,7 +797,12 @@ public int getEncodeSize()
 
          BufferHelper.sizeOfNullableSimpleString(groupID) +
 
-         DataConstants.SIZE_INT; // factoryType
+         DataConstants.SIZE_INT +
```
Comment (gaohoward) [inline hornetq-jms-server/src/main/java/org/hornetq/jms/server/config/impl/ConnectionFactoryConfigurationImpl.java:800]: '+' at the end?

**Pull request merged**